spec/models/exchange_spec.rb

```diff
@@ -275,7 +275,7 @@
 
     ex1 = oc.exchanges.last
     ex1.update_attribute(:tag_list, "wholesale")
-    ex2 = ex1.clone! new_oc
+    ex2 = ex1.reload.clone! new_oc
```

Comment:
I was curious about this, it turns out than when you load/reload the object `tag_list` doesn't get populated till it gets accessed. 
If you were to do the following, the test will pass : 
```ruby
 ex1.reload
 puts ex1.tag_list # <- here tag list gets populated
 ex2 = ex1.reload.clone! new_oc
```

Reply:
Hey @rioug - This is a quite confusing behavior of this gem. Let me explain with the help of screenshots as well:
I queried the last DB record for the `Exchange` and we can see in the object's inspect that the `tag_list` instance variable doesn't get populated.
![image](https://github.com/openfoodfoundation/openfoodnetwork/assets/59338032/370ff75e-f84c-4800-b4b2-f2f6580b1ce3)
However, when I called the tag_list method, it queries the DB to get the tag_list values but does not populate the object's respective instance variable.
![image](https://github.com/openfoodfoundation/openfoodnetwork/assets/59338032/5ee541be-4714-4c93-8407-c8436f7137d5)
When I assign the tag_list the value by update_attribute, we do get it in its value in the object's instance variable as well. The is the Rails behavior of the `update_attribute`, which doesn't reload (_repopulate the object instance variables by querying the DB_):
![image](https://github.com/openfoodfoundation/openfoodnetwork/assets/59338032/9135b59e-f607-4d5e-a657-56a458ec9b99)
When we manually reload the object (_a db query gets called_), we can see that the `tag_list` instance variable gets empty. 
![image](https://github.com/openfoodfoundation/openfoodnetwork/assets/59338032/61e094ae-ae28-401c-bbad-13fd15b61124)
However, when we call the tag_list method, it queries the DB and get the value
![image](https://github.com/openfoodfoundation/openfoodnetwork/assets/59338032/be4f56e6-00b1-46f6-807a-d6a7f6ba4161)
Moreover, we wont get the `tag_list` value in the object's attributes/dup as well because they return/assign the values with respect the instance variables
![image](https://github.com/openfoodfoundation/openfoodnetwork/assets/59338032/319af4f3-9546-4f66-90d9-e6b23e8707ab)

The PR ensures that in the spec, we re-query the DB so that we can ensure the actual behavior which we are facing from the UI.

Please let me know if you have any follow-up questions. Thanks :smile:

Reply:
Thanks for the detailed explanation !